reinforcement_learning/bindings/python/py_api.cc

```diff
@@ -0,0 +1,117 @@
+#include "py_api.h"
+
+#include "config_utility.h"
+#include "configuration.h"
+#include "factory_resolver.h"
+#include "api_status.h"
+#include "err_constants.h"
+
+#include <Python.h>
+
+#include <exception>
+#include <sstream>
+
+namespace reinforcement_learning {
+  namespace python {
+    void dispatch_error_internal(const reinforcement_learning::api_status& status, error_callback* context) {
+      // Obtain global interpreter lock to execute Python.
+      PyGILState_STATE gstate;
+      gstate = PyGILState_Ensure();
+
+      // Callback defined by Python subclass.
+      context->on_error(status.get_error_code(), status.get_error_msg());

```

Comment:
In the way it is being used in live model - no. As the error_callback object is passed in from Python. In general though usages may arise where it could be null so I can put a check in there to be safe